ethereum/client/src/tests.rs

```diff
@@ -0,0 +1,281 @@
+    }
+
+    fn build_contracts<P, T>(
+        provider: P,
+        env: &DeploymentEnv,
+    ) -> Result<Contracts<P, T, Ethereum>, Error>
+    where
+        T: Transport + Clone,
+        P: Provider<T, Ethereum> + Send + Sync + Clone + 'static,
+    {
+        let pk = hex::decode("ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80")
+            .map_err(|_| Error::WrongPrivateKey)?;
+        let contracts =
+            Contracts::new(provider, *env.message_queue_proxy.0, *env.relayer_proxy.0).unwrap();
+        Ok(contracts)
+    }
+
+    pub fn deserialize_merkle_root_json(json_string: &str) -> Result<BlockMerkleRootProof, Error> {
+        let proof: BlockMerkleRootProof = BlockMerkleRootProof::try_from_json_string(json_string)
+            .map_err(|_| Error::WrongJsonFormation)?;
+        Ok(proof)
+    }
+
+    async fn deploy<P, T>(provider: P) -> Result<DeploymentEnv, Error>
+    where
+        T: Transport + Clone,
+        P: Provider<T, Ethereum> + Send + Sync + Clone + 'static,
+    {
+        let vwara_erc20_mock = ERC20Mock::deploy(provider.clone(), "wVARA".to_string())
+            .await
+            .map_err(Error::ErrorDuringContractExecution)?;
+        let verifier_mock = VerifierMock::deploy(provider.clone())
+            .await
+            .map_err(Error::ErrorDuringContractExecution)?;
+
+        let relayer = Relayer::deploy(provider.clone())
+            .await
+            .map_err(Error::ErrorDuringContractExecution)?;
+
+        let erc20_treasury = ERC20Treasury::deploy(provider.clone())
+            .await
+            .map_err(Error::ErrorDuringContractExecution)?;
+
+        let message_queue = MessageQueue::deploy(provider.clone())
+            .await
+            .map_err(Error::ErrorDuringContractExecution)?;
+
+        let relayer_proxy =
+            ProxyContract::deploy(provider.clone(), *relayer.address(), Bytes::new())
+                .await
+                .map_err(Error::ErrorDuringContractExecution)?;
+
+        let message_queue_proxy =
+            ProxyContract::deploy(provider.clone(), *message_queue.address(), Bytes::new())
+                .await
+                .map_err(Error::ErrorDuringContractExecution)?;
+
+        let erc20_treasury_proxy =
+            ProxyContract::deploy(provider.clone(), *message_queue.address(), Bytes::new())
+                .await
+                .map_err(Error::ErrorDuringContractExecution)?;
+
+        Ok(DeploymentEnv {
+            wvara_erc20: *vwara_erc20_mock.address(),
+            verifier: *verifier_mock.address(),
+            message_queue: *message_queue.address(),
+            relayer: *relayer.address(),
+            erc20_treasury: *erc20_treasury.address(),
+            message_queue_proxy: *message_queue_proxy.address(),
+            relayer_proxy: *relayer_proxy.address(),
+            erc20_treasury_proxy: *erc20_treasury_proxy.address(),
+        })
+    }
+
+    fn build_merkle_proof_json() -> String {
```

Comment:
Just construct `BlockMerkleRootProof` directly without parsing and deserializing it

Reply:
done